### PR TITLE
ENH: two algorithms for the convex hull problem of a set of 2d points on a plain

### PR DESCRIPTION
**Pull request by maxwell-aladago**
1. A naive brute-force algorithm
2. A divide-and-conquer algorithm

### CODE REVIEW

File: divide_and_conquer/convex_hull.py [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    ValueError: x and y must be both numeric types but got <class 'str'>, <class 'str'> instead
+     """
+
+    def __init__(self, x, y):
+        if not (isinstance(x, Number) and isinstance(y, Number)):
+            try:
+                x, y = float(x), float(y)
+            except ValueError as e:
+                e.args = ("x and y must be both numeric types "
+                          "but got {}, {} instead".format(str(type(x)), str(type(y))), )
```
Comment (cclauss) [inline divide_and_conquer/convex_hull.py]: Are the __str()_ calls required in this context?

File: divide_and_conquer/convex_hull.py [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+
+    def __le__(self, other):
+        if self.x < other.x:
+            return True
+        elif self.x == other.x:
+            return self.y <= other.y
+        return False
+
+    def __repr__(self):
+        return "({}, {})".format(self.x,  self.y)
```
Comment (cclauss) [inline divide_and_conquer/convex_hull.py]: Try this one with an f”string” and see if it is more concise.

**Comment by maxwell-aladago** [comment]
Actually no, forgot to remove them before committing

File: divide_and_conquer/convex_hull.py [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+     [(0, 0), (1, 0), (10, 1)]
+     >>> convex_hull_bf([[0, 0], [1, 0], [10, 0]])
+     [(0, 0), (10, 0)]
+     >>> convex_hull_bf([[-1, 1],[-1, -1], [0, 0], [0.5, 0.5], [1, -1], [1, 1], [-0.75, 1]])
+     [(-1, -1), (-1, 1), (1, -1), (1, 1)]
+     >>> convex_hull_bf([(0, 3), (2, 2), (1, 1), (2, 1), (3, 0), (0, 0), (3, 3), (2, -1), (2, -4), (1, -3)])
+     [(0, 0), (0, 3), (1, -3), (2, -4), (3, 0), (3, 3)]
+    """
+
+    points = _validate_input(points)
```
Comment (cclauss) [inline divide_and_conquer/convex_hull.py]: points = sorted(_validate_input(points))

File: divide_and_conquer/convex_hull.py (hunk 2 above)
Comment (maxwell-aladago) [reply divide_and_conquer/convex_hull.py]: But f'strings' don't work with python 2 or?

**Comment by maxwell-aladago** [comment]
> Impressive!

Thanks, I will make the suggested changes soon.

**Pull request merged**